### PR TITLE
Docs docker

### PR DESCRIPTION
**Pull request by kingthorin**
- Migrate baseline, full, and api packaged scan details from GitHub wiki.
- Migrate webswing details from GitHub wiki.
- Migrate scan hooks from zaproxy code repo.
- In the api doc I dropped the word "new" from the sentence "It also includes 2 new scripts that:"
- Layout "links" now includes content ahead of the list of links, to allow for a "lead-in" paragraph etc. (This may cause extra blank lines to be added to pages which use the same layout and have no content.)
- Update info reference in d_docker.yml.
- Added a layout docker/single.html which has a specific breadcrumb.
- Tweaked various addon, blog, and FAQ references.
- Remove references to SOAP in api-scan info.
- Add 'unused' param to scan hook 'wrap' functions.

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

**Comment by psiinon** [comment]
Will also need to update the Downloads page: https://www.zaproxy.org/download/#docker

File: site/content/docs/docker/baseline.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
+---
+title: "ZAP - Baseline Scan"
+tags: 
+- docker
+- guide
+- packaged_scan
+type: page
+---
+
+The ZAP Baseline scan is a script that is available in the ZAP [[Docker](../docker)] images.
```
Comment (psiinon) [inline site/content/docs/docker/baseline.md:10]: Extra '[' needs removing.

File: site/content/docs/docker/_index.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+---
+title: "ZAP Docker Documentation"
+type: page
+layout: links
```
Comment (psiinon) [inline site/content/docs/docker/_index.md:4]: I'd be tempted to add a short paragraph introducing the docker images - eg they are an easy way to run ZAP, especially in a CI/CD environment.

**Comment by kingthorin** [comment]
Now includes update to docker download info.

**Comment by thc202** [comment]
Thank you!

File: site/content/docs/docker/api-scan.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+tags: 
+- docker
+- guide
+- packaged_scan
+type: docker
+---
+
+The ZAP API scan is a script that is available in the ZAP [Docker](../about/) images.
+
+It is tuned for performing scans against APIs defined by either OpenAPI or SOAP via either a local file or a URL.
```
Comment (psiinon) [inline site/content/docs/docker/api-scan.md:12]: We should probably remove SOAP, or say that its not currently supported :/

File: site/content/docs/docker/scan-hooks.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+- `importing_openapi(target_url, target_file)`
+- `importing_soap(target_url, target_file)`
+- `load_config(config, config_dict, config_msg, out_of_scope_dict)`
+- `print_rules_wrap(count, inprog_count)`
+- `start_zap(port, extra_zap_params)`
+- `start_docker_zap(docker_image, port, extra_zap_params, mount_dir)`
+- `start_docker_zap_wrap(cid)`
+- `zap_access_target(zap, target)`
+- `zap_spider(zap, target)`
+- `zap_spider_wrap()`
```
Comment (psiinon) [inline site/content/docs/docker/scan-hooks.md:67]: I found out the hard way that this method actually has one param and the script will fail if its not defined. I dont know what its for and when I printed it out its 'null' - could well be a bug but we should document this behaviour - could be an `unused` param?

**Comment by psiinon** [comment]
A couple of minor tweaks required, otherwise looks good :)

**Pull request merged**